src/System.Text.Utf8/src/System/Text/Utf8/Utf8EncodedCodePoint.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Runtime.InteropServices;
+
+namespace System.Text.Utf8
+{
+    // TODO: Remove this class, make people use char.IsSurrogate and char.ToUtf32 to convert to code points and use Utf8Encoder to write bytes
+    [StructLayout(LayoutKind.Explicit)]
+    public struct Utf8EncodedCodePoint
+    {
+        public unsafe Utf8EncodedCodePoint(char character) : this()
+        {
+            if (char.IsSurrogate(character))
+                throw new ArgumentOutOfRangeException("character", "Surrogate characters are not allowed");
```

Comment:
I added TODO to validate these constructors. I will probably end up removing this class at one point.